### PR TITLE
Introduce namespacer

### PR DESCRIPTION
**Pull request by dstiliadis**
Introduces a namespacer interface that enables a user to customize the source of the
namespace information. Users of the library can introduce their own mechanisms 
for retrieving the namespace from other headers, JWT tokens, or anything else available
in an HTTP request.

### CODE REVIEW

File: request_config.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+// SetNamespacer will configure the package. It must be only called once
+// and it is global for the package.
+func SetNamespacer(custom Namespacer) {
+	namespacer = custom
+}
+
+// ExtractNamespace extracts the namespace value from an http.Request.
+func ExtractNamespace(r *http.Request) (string, error) {
```
Comment (primalmotion) [inline request_config.go:39]: I don't think this should be public

File: request_config.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	return namespacer.Extract(r)
+}
+
+// InjectNamespace will configure the http request with the right namespace value.
+func InjectNamespace(r *http.Request, namespace string) error {
```
Comment (primalmotion) [inline request_config.go:44]: same, I don't think this should be public

File: namespacer.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+func (d *defaultNamespacer) Extract(r *http.Request) (string, error) {
+	return r.Header.Get("X-Namespace"), nil
+}
+
+// defaultInjector will set the namespace as an HTTP header.
+func (d *defaultNamespacer) Inject(r *http.Request, namespace string) error {
+	if r.Header == nil {
+		r.Header = http.Header{}
+	}
+	r.Header.Add("X-Namespace", namespace)
```
Comment (primalmotion) [inline namespacer.go:28]: This shoukd use .Set not .Add

Comment (dstiliadis) [reply namespacer.go]: fixed

File: request.go [hunk 4]
```diff
@@ -282,9 +282,14 @@ func NewRequestFromHTTPRequest(req *http.Request, manager ModelManager) (*Reques
 		clientIP = req.RemoteAddr
 	}
 
+	var namespace string
+	if namespace, err = GetNamespacer().Extract(req); err != nil {
```
Comment (primalmotion) [inline request.go:286]: If namespacer is nil, this panics.

**Pull request merged**